### DISCUSSION TITLE
The service "Pimcore\Log\Handler\ApplicationLoggerDb" has a dependency on a non-existent service "monolog.processor.psr_log_message"

### DISCUSSION THREAD

**Discussion by ghost** [pimcore/pimcore#9243] [Q&A / Support]
Hi,

With a custom environment, I run into a problem that was once fixed. Namely:
```
The service "Pimcore\Log\Handler\ApplicationLoggerDb" has a dependency on a non-existent service 
monolog.processor.psr_log_message"
```
The problem was raised here: https://github.com/pimcore/pimcore/issues/2481 and in short states the following: If you create a custom environment (for instance "phpunit" or "behat") the "monolog.processor.psr_log_message" service is not available. The problem was fixed here https://github.com/pimcore/pimcore/commit/e2e2315c54f121573c7b21b723e1be55ccd0c4cc

However, it was later broken with a seemingly unrelated commit:
```
1904d511d14564185fa6e5269a8922c2f9896734

Author: Sebastian Blank <blank@data-factory.net>
Date:   Mon May 3 16:18:57 2021 +0200

    Fix PHPStan level 4 errors (#8906)

```
With seeminly unrelated, I mean that fixing PHPStan Errors should not result in the deletion of a class that actually did something. I can of course create a fix for this, but I would like to know how to move forward. Simply cherry picking the solution in https://github.com/pimcore/pimcore/commit/e2e2315c54f121573c7b21b723e1be55ccd0c4cc indeed fixes the problem.

**Comment by dvesh3** [comment]
@davidmaes please create a PR which reverts the deletion. thanks!

**Reply by ghost** [reply]
cool :)
